src/js/d3ViewComponent.js

```diff
@@ -0,0 +1,94 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    // The D3 view component that is able to:
+    // 1. Convert jQuery DOM elements to D3 elements;
+    // 2. Attach D3 DOM event listeners;
+    // 3. Synthesize that.options.styles and that.options.selectors to combine elements with the same key into that.classes
+
+    fluid.defaults("gpii.d3ViewComponent", {
+        gradeNames: ["fluid.viewComponent", "autoInit"],
+        members: {
+            classes: {
+                expander: {
+                    funcName: "gpii.d3ViewComponent.synthesizeClasses",
+                    args: ["{that}.options.styles", "{that}.options.selectors"]
+                }
+            }
+        },
+        invokers: {
+            jQueryToD3: {
```

Comment:
I like this - basically, provide d3-namespaced extensions to the common Infusion locator functions, which internally would use the jQueryToD3 function. Implementation code is already doing this in various places anyway.